spec/requests/flavors_spec.rb

```diff
@@ -0,0 +1,76 @@
+RSpec.describe 'Flavors API' do
+  let(:flavor) { FactoryGirl.create(:flavor_openstack) }
+
+  describe 'GET/api/flavors' do
+    it 'lists all the flavor bases with an appropriate role' do
+      flavor = FactoryGirl.create(:flavor_openstack)
+
+      api_basic_authorize collection_action_identifier(:flavors, :read, :get)
+
+      run_get(api_flavors_url)
+
+      expected = {
+        'count'     => 1,
+        'subcount'  => 1,
+        'name'      => 'flavors',
+        'resources' => [hash_including('href' => a_string_matching(api_flavors_url(nil, flavor.compressed_id)))]
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'forbids access to flavor bases without an appropriate role' do
+      api_basic_authorize
+
+      run_get(api_flavors_url)
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe 'GET /api/flavors/:id' do
+    it 'will show a flavor base' do
+      api_basic_authorize action_identifier(:flavors, :read, :resource_actions, :get)
+
+      flavor = FactoryGirl.create(:flavor)
+
+      run_get(api_flavor_url(nil, flavor))
+
+      expected = {
+        'href' => a_string_matching(api_flavors_url(nil, flavor.compressed_id))
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'forbids access to a flavor base' do
+      api_basic_authorize
+
+      run_get(api_flavors_url)
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe 'DELETE /api/flavors/:id' do
+    it 'will delete a flavor' do
+      flavor = FactoryGirl.create(:flavor)
+
+      api_basic_authorize action_identifier(:flavors, :delete, :resource_actions, :delete)
+
+      run_delete api_flavor_url(nil, flavor)
+
+      expect(response).to have_http_status(:no_content)
+    end
+
+    it 'will not delete a flavor without an appropriate role' do
+      flavor = FactoryGirl.create(:flavor)
+
+      api_basic_authorize
+
+      run_delete api_flavors_url(nil, flavor)
```

Comment:
I believe you want `api_flavor_url(nil, flavor) here

Reply:
@jntullo Silly mistake :) Thank you, now it`s ok!